xray/segment.go

```diff
@@ -352,6 +352,11 @@ func (seg *Segment) Close(err error) {
 		return
 	}
 
+	if seg == nil {
+		logger.Debugf("No input subsegment to end. No-op")
```

Comment:
Is there a way we can delegate this situation to be handled by the ContextMissingStrategy?
This way it can be just a log or a runtime exception if a user choses to.

Reply:
I talked with Will in regards to its implementation in Java.

"I think that’s the expected behavior for this method. If a customer explicitly passes in the subsegment to be closed, then we shouldn’t throw context missing errors because we’re not interacting with the context, we just use the subsegment provided by the customer as noted in [javadocs](https://github.com/aws/aws-xray-sdk-java/blob/fd93c8a938e635137dedc70edf9b709f039ba0c1/aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/AWSXRayRecorder.java#L567-L568)."